File: Array (MS)/128_Longest_Consecutive_Sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
  if (nums.length == 0) {
    return 0;
  }

  nums.sort(function (a, b) {
    return a - b;
  });

  console.log("Sorted version == ", nums);

  var count = 1;
  var result = 1;

  for (let i = 1; i < nums.length; i++) {
    if (nums[i - 1] === nums[i]) {
      continue;
    }
    nums[i - 1] - nums[i] === -1 ? (count += 1) : (count = 1);
    if (count > result) {
      result = count;
    }
  }

  return result;
};

// Runtime - 101ms      | Beats 83.11% of users with JavaScript
// Memory - 51.36mb     | Beats 97.84% of users with JavaScript

File: Array (TVS)/128_Longest_Consecutive_Sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0

        nums.sort()

        count = 1
        maxValue = 0

        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                if nums[i] == nums[i - 1] + 1:
                    count += 1
                else:
                    maxValue = max(count, maxValue)
                    count = 1

        return max(maxValue, count)

if __name__ == "__main__":
  solution = Solution()
  print(solution.longestConsecutive([9,1,4,7,3,-1,0,5,8,-1,6]))